### DISCUSSION TITLE
dragging multiple 3D scene objects together as one using scene groups

### DISCUSSION THREAD

**Discussion by h0uter** [zauberzeug/nicegui#3551] [Ideas / Feature Requests]
Now that it is possible to handle events during dragging in `1.4.35` what would be the recommended approach for dragging certain scene objects together as one?  If we use [the dragging example from the docs](https://nicegui.io/documentation/scene#subscribe_to_the_drag_event) to implement a 3D "selection" of some sort it would be useful if we can also drag the entire selection as one.

The first thing I thought of was to use the scene group. I expected this would allow me to lock a set of scene objects together in a scene group and use that composite scene group just as if it were a regular individual scene object. Calling `.move()`  on a scene group moves all members together and this works great. However in relation to dragging I ran into some challenges.

1. I haven't found a way to drag a scene group as one entity. calling `.draggable()` on the scene group just makes all members draggable individually. This means that to extend [the example](https://nicegui.io/documentation/scene#subscribe_to_the_drag_event) we need to check which object of the scene group is being dragged and update the other group members positions in the drag event callback.
2. When checking if one of the group members is being dragged I haven't found an easy way to check against the list of  `.id`s of all members of the group. The group gets its own id, but there is no `.ids` for its members.

I am creating this discussion / feature request to think about what would be the recommended approach for realizing something like this and share how far I was able to come.

https://github.com/user-attachments/assets/c25296f4-12ff-4747-97bb-5e130606bb02

In this vid we can either drag the yellow sphere or the yelow circle surrounding it and move them together as one.

```python
import math

from nicegui import events, ui

with ui.scene(drag_constraints="z=0") as scene:
    box_height = 0.5
    box = (
        scene.box(1, 1, box_height, wireframe=True)
        .move(0, 0, box_height / 2)
        .material("Orange")
    )

    with scene.group() as anchor_group:
        anchor_color = "yellow"
        sphere1 = scene.sphere(0.2).material(anchor_color).move(0.5, -0.5)
        circle1 = (
            scene.cylinder(0.5, 0.5, 0, radial_segments=30)
            .rotate(r_x=math.pi / 2, r_y=0, r_z=0)
            .material(anchor_color)
            .move(0.5, -0.5)
        )

    anchor_group.draggable()

    sphere2 = scene.sphere(0.2).move(-0.5, 0.5).material("SteelBlue").draggable()

    distractor_sphere = scene.sphere(0.2).move(-1, 1).material("red").draggable()


def handle_drag(e: events.GenericEventArguments) -> None:
    if e.args["object_id"] not in [sphere1.id, sphere2.id, circle1.id]:
        return

    x1 = sphere1.x if e.args["object_id"] == sphere2.id else e.args["x"]
    y1 = sphere1.y if e.args["object_id"] == sphere2.id else e.args["y"]

    # it would be nice if we could use: `if ... in anchor_group.ids`
    x2 = sphere2.x if e.args["object_id"] in [sphere1.id, circle1.id] else e.args["x"]
    y2 = sphere2.y if e.args["object_id"] in [sphere1.id, circle1.id] else e.args["y"]

    box.move((x1 + x2) / 2, (y1 + y2) / 2, box_height / 2).scale(x2 - x1, y2 - y1, 1)

    """here we manually update the other members of the scene group"""

    # note that just using `anchor_group.move(x1,y1)` wont work as desired
    # anchor_group.move(x1, y1)

    if e.args["object_id"] == sphere1.id:
        circle1.move(x1, y1)
    elif e.args["object_id"] == circle1.id:
        sphere1.move(x1, y1)


scene.on("drag", handle_drag)

ui.run()
```

**Comment by falkoschindler** [comment]
Hi @h0uter,

We just merged PR #3549 into main, which fixes dragging of groups. Does that resolve your problem?
